tests/tensorflow_autolog/test_tensorflow2_autolog.py

```diff
@@ -0,0 +1,175 @@
+# pep8: disable=E501
+
+from __future__ import print_function
+
+import collections
+import shutil
+import pytest
+import tempfile
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+import mlflow
+import mlflow.tensorflow
+import mlflow.keras
+
+import os
+
+SavedModelInfo = collections.namedtuple(
+    "SavedModelInfo",
+    ["path", "meta_graph_tags", "signature_def_key", "inference_df", "expected_results_df"])
+
+client = mlflow.tracking.MlflowClient()
+
+
+@pytest.fixture
+def random_train_data():
+    return np.random.random((1000, 32))
+
+
+@pytest.fixture
+def tf_keras_random_data_run(random_train_data):
+    mlflow.tensorflow.autolog(every_n_iter=5)
+
+    def random_one_hot_labels(shape):
+        n, n_class = shape
+        classes = np.random.randint(0, n_class, n)
+        labels = np.zeros((n, n_class))
+        labels[np.arange(n), classes] = 1
+        return labels
+
+    with mlflow.start_run() as run:
+        data = random_train_data
+        labels = random_one_hot_labels((1000, 10))
+
+        model = tf.keras.Sequential()
+
+        model.add(layers.Dense(64, activation='relu', input_shape=(32,)))
+        model.add(layers.Dense(64, activation='relu'))
+        model.add(layers.Dense(10, activation='softmax'))
+
+        model.compile(optimizer=tf.keras.optimizers.Adam(),
+                      loss='categorical_crossentropy',
+                      metrics=['accuracy'])
+
+        model.fit(data, labels, epochs=10)
+
+    return client.get_run(run.info.run_id)
+
+
+@pytest.mark.large
+def test_tf_keras_autolog_logs_expected_data(tf_keras_random_data_run):
+    data = tf_keras_random_data_run.data
+    assert 'accuracy' in data.metrics
+    assert 'loss' in data.metrics
+    assert 'name' in data.params
```

Comment:
Hm having a param called `'name'` is a bit confusing out of context, maybe we could preface each of the optimizer param names with `optimizer_`? So we'd have `optimizer_name`, `optimizer_learning_rate`, ... etc. If that results in the param name being truncated in the UI (would be good to try it out & confirm), maybe we could do `opt_name`, `opt_learning_rate`, etc instead.